modules/EvseSlac/main/evse_fsm.cpp

```diff
@@ -355,6 +356,8 @@ void EvseFSM::sd_matching_hsm(FSMContextType& ctx, const EventSlacMessage& ev) {
     auto& start_atten = msg_in.get_payload<slac::messages::cm_start_atten_char_ind>();
 
     if (!matching_ctx.conforms(msg_in.get_src_mac(), start_atten.run_id)) {
+        EVLOG_warning << fmt::format("Ignoring CM_START_ATTEN_CHAR from {}, RunID {}",
```

Comment:
Please take a look at all these `const char*` casts, most of the the buffers are not required to be null-terminated, like `get_src_mac` for example, or even the `run_id`.  This could lead to segfaults.  Furthermore, the c++ way of casting pointers should be `reinterpret_cast<const char*>(pointer)`.  You probably need to write a separate printing helper for these values. The source mac should be represented as hex anyway

Reply:
I removed printing of the details, it is more important to see that something was dropped at all.